delivery/media/watermarking/lib/media-delivery-watermarking.js

```diff
@@ -0,0 +1,2134 @@
+import { WritableStream } from "streams";
+
+import { crypto, pem2ab } from "crypto";
+
+import { TextEncoder as TextEncoder$1, base64url, base16, atob } from "encoding";
+
+import { logger } from "log";
+
+import { IrdetoAlgorithm } from "irdeto-algorithm.js";
```

Comment:
I don't see this file in the commit. Is it already in the repo?

Reply:
@evan-hughes : This is what we will provide out of band to the interested customer who wants to use the module.

Reply:
Would it make sense to generalize the name now so we don't have vendor names in our published modules?

Reply:
I chatted to some folks about importing vendor code. Rather than importing the vendor module in our library, it would be cleaner if the customer imported the vendor code and passed the `generateTmid` function into our library.  Something along the lines of dependency injection through the config object. 

That would give the customer more control over the vendor code and allows us to get the vendor name out of our public module.

Reply:
True, but in direct case the logic to generate TMID is simple decryption and can be coded in the watermarking module itself. So are we saying we accept the vendor code (function to generate tmid) as an optional argument (i.e. should be optional due to direct case ) or for direct case we expect customer to pass the decryption logic to generate TMID.

How about accepting generated TMID itself as a parameter? . i.e customer uses the vendor code,  generate the tmid and pass it to our module? So in direct case its the customer responsibility to decrypt it and pass the generated TMID?

Reply:
>  I don't follow. The TMID function appears to take a number of decoded token fields as arguments. Would the customer need to pass a function that would be called on those arguments?
The arguments required for tmid calculated is from decoded token itself and validateToken function returns the decoded data after verification.
customer will first validate the token and then pass the required args from token to TMID function and generate the TMID.  The generated TMID can be passed in getWMPathWithVariant function call.